### ISSUE TITLE
heka_tcp output plugin terminates if server is not listening anymore

### ISSUE THREAD

**Issue by bbinet** [mozilla-services/lua_sandbox_extensions#65]
Today I noticed that if I temporarily stop my heka_tcp input plugin, then the heka_tcp ouput plugin that were sending data to that stopped plugin will terminate quickly with the following error in logs:
```
1484245022748499356 [error] output.heka_tcp terminated: process_message() instruction_limit exceeded
```

Note that my heka_tcp input/output plugin are configured with ssl support.

**Comment by trink** [comment]
I had one hang (possibly in socket.write(), process_message never returned) under very similar conditions.  I was working with Ops and we were able to reproduce it without SSL but got no further.  I am assuming this failure is related and expect to find something that stepped on memory (as there is nothing that would exhaust the instruction limit in that plugin).

**Comment by bbinet** [comment]
In fact it actually enters an endless loop because `client:send`returns error "Broken pipe", so the plugin tries a new `send_message(msg, i)`:
https://github.com/mozilla-services/lua_sandbox_extensions/blob/master/socket/sandboxes/heka/output/heka_tcp.lua#L70

Patch to come.

**Comment by bbinet** [comment]
This is addressed in pull request #66

**Comment by trink** [comment]
Good catch, I just looked at are prod config and the instruction_limit was set to zero so that explains our hang as it was just infinitely looping.
